app/controllers/api/storage.php

```diff
@@ -21,8 +22,134 @@
+            ->setParam('resource', 'database/collection/' . $data->getId())
+            ->setParam('data', $data->getArrayCopy())
+        ;
+
+        $response->setStatusCode(Response::STATUS_CODE_CREATED);
+        $response->dynamic2($data, Response::MODEL_BUCKET);
+    });
+
+App::get('/v1/storage/buckets')
+    ->desc('List buckets')
+    ->groups(['api', 'storage'])
+    ->label('scope', 'buckets.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_SESSION, APP_AUTH_TYPE_KEY, APP_AUTH_TYPE_JWT])
+    ->label('sdk.namespace', 'storage')
+    ->label('sdk.method', 'listBuckets')
+    ->label('sdk.description', '/docs/references/storage/list-buckets.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_BUCKET_LIST)
+    ->param('search', '', new Text(256), 'Search term to filter your list results. Max length: 256 chars.', true)
+    ->param('limit', 25, new Range(0, 100), 'Results limit value. By default will return maximum 25 results. Maximum of 100 results allowed per request.', true)
+    ->param('offset', 0, new Range(0, 2000), 'Results offset. The default value is 0. Use this param to manage pagination.', true)
+    ->param('orderType', 'ASC', new WhiteList(['ASC', 'DESC'], true), 'Order result by ASC or DESC order.', true)
+    ->inject('response')
+    ->inject('dbForInternal')
+    ->action(function ($search, $limit, $offset, $orderType, $response, $dbForInternal) {
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Utopia\Database\Database $dbForInternal */
+
+        $queries = ($search) ? [new Query('name', Query::TYPE_SEARCH, $search)] : [];
+
+        $response->dynamic2(new Document([
+            'buckets' => $dbForInternal->find('buckets', $queries, $limit, $offset, ['_id'], [$orderType]),
+            'sum' => $dbForInternal->count('buckets', $queries, APP_LIMIT_COUNT),
+        ]), Response::MODEL_BUCKET_LIST);
+    });
+
+App::get('/v1/storage/buckets/:bucketId')
+    ->desc('Get Bucket')
+    ->groups(['api', 'storage'])
+    ->label('scope', 'buckets.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_SESSION, APP_AUTH_TYPE_KEY, APP_AUTH_TYPE_JWT])
+    ->label('sdk.namespace', 'storage')
+    ->label('sdk.method', 'getBucket')
+    ->label('sdk.description', '/docs/references/storage/get-bucket.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_BUCKET)
+    ->param('bucketId', '', new UID(), 'Bucket unique ID.')
+    ->inject('response')
+    ->inject('dbForInternal')
+    ->action(function ($bucketId, $response, $dbForInternal) {
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Utopia\Database\Database $dbForInternal */
+
+        $bucket = $dbForInternal->getDocument('buckets', $bucketId);
+
+        if (empty($bucket->getId())) {
```

Comment:
```suggestion
        if ($bucket->isEmpty()) {
```

I think we aren't consistent enough in the code base, but this would probably be a cleaner approach.